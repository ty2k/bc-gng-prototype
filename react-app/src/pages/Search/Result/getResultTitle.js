export default function getResultTitle(result, tab) {
  // All (1) or Services (4) tabs
  if (tab === 1 || tab === 4) {
    return (
      <a className="title" href={result?.UE}>
        {result?.T?.toString().split(" - Province of British Columbia", 1)[0]}
      </a>
    );
    // News (2) tab
  } else if (tab === 2) {
    return (
      <a className="title" href={result?.UE}>
        {result?.T?.toString().split(" | BC Gov News", 1)[0]}
      </a>
    );
    // Documents (3) tab and default
  } else {
    return (
      <a className="title" href={result?.UE}>
        {result?.T}
      </a>
    );
  }
}
