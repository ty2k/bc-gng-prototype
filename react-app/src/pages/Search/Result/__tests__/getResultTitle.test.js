import TestRenderer from "react-test-renderer";

import getResultTitle from "../getResultTitle";

import {
  mockResultTab1,
  mockResultTab2,
  mockResultTab3,
  mockResultTab4,
} from "../__mocks__/mockResultData";

describe("getResultTitle", () => {
  const titleTab1 = TestRenderer.create(
    getResultTitle(mockResultTab1, 1)
  ).toJSON();
  const titleTab2 = TestRenderer.create(
    getResultTitle(mockResultTab2, 2)
  ).toJSON();
  const titleTab3 = TestRenderer.create(
    getResultTitle(mockResultTab3, 3)
  ).toJSON();
  const titleTab4 = TestRenderer.create(
    getResultTitle(mockResultTab4, 4)
  ).toJSON();

  it("renders a Tab 1 (All) title correctly", () => {
    expect(titleTab1).toMatchSnapshot();
  });
  it("renders a Tab 2 (All) title correctly", () => {
    expect(titleTab2).toMatchSnapshot();
  });
  it("renders a Tab 3 (All) title correctly", () => {
    expect(titleTab3).toMatchSnapshot();
  });
  it("renders a Tab 4 (All) title correctly", () => {
    expect(titleTab4).toMatchSnapshot();
  });
});
