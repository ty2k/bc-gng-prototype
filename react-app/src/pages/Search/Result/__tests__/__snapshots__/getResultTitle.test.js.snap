// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getResultTitle renders a Tab 1 (All) title correctly 1`] = `
<a
  className="title"
  href={
    Array [
      "https://www2.gov.bc.ca/gov/content/careers-myhr/all-employees/pay-benefits/benefits/msp",
    ]
  }
>
  Medical Services Plan (MSP)
</a>
`;

exports[`getResultTitle renders a Tab 2 (All) title correctly 1`] = `
<a
  className="title"
  href={
    Array [
      "https://news.gov.bc.ca/releases/2019HLTH0114-001555",
    ]
  }
>
  International students continue to contribute to B.C. health care as MSP premiums eliminated
</a>
`;

exports[`getResultTitle renders a Tab 3 (All) title correctly 1`] = `
<a
  className="title"
  href={
    Array [
      "https://www2.gov.bc.ca/assets/gov/health/practitioner-pro/medical-services-plan/msp-service-locations-codes.pdf",
    ]
  }
>
  msp-service-locations-codes.pdf
</a>
`;

exports[`getResultTitle renders a Tab 4 (All) title correctly 1`] = `
<a
  className="title"
  href={
    Array [
      "https://www2.gov.bc.ca/gov/content/health/health-drug-coverage/msp/bc-residents",
    ]
  }
>
  Medical Services Plan (MSP) for British Columbia (B.C.) Residents
</a>
`;
